src/main/java/org/breedinginsight/brapi/v2/BrAPIStudiesController.java

```diff
@@ -82,11 +96,28 @@ public HttpResponse<Response<DataResponse<List<BrAPIStudy>>>> getStudies(
             @QueryValue @QueryValid(using = StudyQueryMapper.class) @Valid StudyQuery queryParams) {
         try {
             log.debug("fetching studies for program: " + programId);
+            List<BrAPIStudy> studies;
+
+            Optional<ProgramUser> experimentalCollaborator = programUserService.getIfExperimentalCollaborator(programId, securityService.getUser().getId());
+            // If the program user is an experimental collaborator, filter results.
+            if (experimentalCollaborator.isPresent()) {
+                Optional<Program> program = programService.getById(programId);
+                if (program.isEmpty()) {
+                    return HttpResponse.notFound();
+                }
+
+                List<UUID> experimentIds = experimentalCollaboratorService.getAuthorizedExperimentIds(experimentalCollaborator.get().getId());
+                studies = studyService.getStudiesByExperimentIds(program.get(), experimentIds)
+                        .stream()
+                        .peek(this::setDbIds)
+                        .collect(Collectors.toList());
+            } else {
+                studies = studyService.getStudies(programId)
+                        .stream()
+                        .peek(this::setDbIds)
+                        .collect(Collectors.toList());
+            }
```

Comment:
Optional methods could be used to refactor the fetching logic.
```suggestion
  Optional<HttpResponse> httpResponse = programUserService.getIfExperimentalCollaborator(programId, securityService.getUser().getId())
        .flatMap(experimentalCollaborator -> {  //If the program user is an experimental collaborator, filter results.
            Optional<Program> program = programService.getById(programId);
            return program.map(p -> {
                List<UUID> experimentIds = experimentalCollaboratorService.getAuthorizedExperimentIds(experimentalCollaborator.getId());
                List<Study> studies = studyService.getStudiesByExperimentIds(p, experimentIds)
                        .stream()
                        .peek(this::setDbIds)
                        .collect(Collectors.toList());
                return HttpResponse.ok(studies);
            });
        })
        .orElseGet(() -> { //Otherwise, fallback to fetching studies without considering collaborator status
            List<Study> studies = studyService.getStudies(programId)
                    .stream()
                    .peek(this::setDbIds)
                    .collect(Collectors.toList());
            return HttpResponse.ok(studies);
        });

return httpResponse.orElse(HttpResponse.notFound());
```

Reply:
I tried to test this and the BrAPITrialsController suggestions locally, they both resulted in build errors. Are there problems in my code as written that you would like me to address?